src/EnergyPlus/EMSManager.cc

```diff
@@ -1756,25 +1712,36 @@ namespace EMSManager {
                 auto &nodeSetpointCheck = state.dataLoopNodes->NodeSetpointCheck(NodeNum);
                 nodeSetpointCheck.needsSetpointChecking = true;
 
-                auto const SELECT_CASE_var(SetPointType);
-                if (SELECT_CASE_var == SPControlType::TemperatureSetPoint) {
+                switch (SetPointType) {
```

Comment:
If you look at `struct NodeSetpointCheckData` in `DataLoopNode.hh`, the fields of that struct can be changed to a `std::array<bool, static_cast<int>(SPControlType::Num)> checkSetPoint`. This would reduce the entire `switch` to the line `nodeSetpointCheck.checkSetPoint[static_cast<int>(SetPointType)] = true;`. 

It's one thing to change `SELECT_CASE_var` `if-else` logic to `switch`, but the "next-level" refactor is to see if the underlying data structure(s) can be made a `std::array` that is indexed by the `enum class` which would eliminate the need for control-flow at all and replace it with data lookups. It takes a bit of practice, yes, but once you learn to see it, it's a powerful pattern.